dialect/sql/builder.go

```diff
@@ -1536,7 +1536,7 @@ func In(col string, args ...interface{}) *Predicate {
 // In appends the `IN` predicate.
 func (p *Predicate) In(col string, args ...interface{}) *Predicate {
 	if len(args) == 0 {
-		return p
+		return p.False()
```

Comment:
```suggestion
		// If no arguments were provided, append the FALSE constants,
		// since we can't apply "IN ()". This will make this predicate falsy.
		return p.False()
````

Reply:
thanks added in [f1fcc1c](https://github.com/ent/ent/pull/2735/commits/f1fcc1c9238a3b82ecdcf5e32246007641ab46f3) 👍